docs/USAGE-GUIDE.md

```diff
@@ -830,63 +830,23 @@ A PSK must not be shared between more than one server and one client. An entity
 
 ### Configuring External Pre-Shared Keys
 
-Use the following APIs to configure external pre-shared keys.
+Both clients and servers will need to create and append a PSK to a connection in order to negotiate that PSK. Call `s2n_external_psk_new()` to allocate memory for a PSK object. Call `s2n_psk_set_identity()` to set a unique identifier for the PSK. Note that this identity will be transmitted over the network unencrypted, so do not include any confidential information in it. Call `s2n_psk_set_secret()` to set the secret value for a given PSK. Deriving a shared secret from a password or other low-entropy source is _not_ secure and is subject to dictionary attacks. Call `s2n_psk_set_hmac()` to change the default PSK hmac algorithm (**S2N_PSK_HMAC_SHA256**) for a given PSK. Note that the hmac algorithm may influence server cipher-suite selection.
```

Comment:
I think that's outside of the scope here. We do provide some guidelines about what not to do. I would rather not get into writing docs on how to generate a secure PSK since s2n won't do that for you, seems like people should look to the libcrypto they're using for guidelines there.

Reply:
Don't we provide minimal guidlines for STEKs? Maybe we should provide the same level for PSKs.

Reply:
Yeah I mean, in the session resumption section we direct users to a RFC which provides some guidelines on [secrets](https://www.rfc-editor.org/rfc/rfc5077#section-5.5). Should I just link to that? My hesitation is that that RFC was written for ticket keys, so it doesn't exactly fit this usecase.

Reply:
Looks like there's a PSK version: https://www.rfc-editor.org/rfc/rfc9257.html#name-recommendations-for-externa
That's a really written informational document we should definitely link somewhere.

Reply:
😮  I've never seen this before, very nice.